pkg/yurthub/otaupdate/ota.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package otaupdate
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+const (
+	PodUpdatableAnnotation = "apps.openyurt.io/pod-updatable"
+)
+
+type PodStatus struct {
+	Namespace string
+	PodName   string
+	// TODO: whether need to display
+	// OldImgs    []string
+	// NewImgs    []string
+	Updatable bool
+}
+
+// GetPods return all daemonset pods' update information by PodStatus
+func GetPods(clientset kubernetes.Interface) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		podsStatus, err := getPodUpdateStatus(clientset)
```

Comment:
we need to return `PodList` for `/pods` request, not to define a new struct `PodsStatus`.